specs/phase1/validator.md

```diff
@@ -251,24 +251,67 @@ A validator should create and broadcast the `attestation` to the associated atte
 - Let `head_block` be the result of running the fork choice during the assigned slot.
 - Let `head_state` be the state of `head_block` processed through any empty slots up to the assigned slot using `process_slots(state, slot)`.
 - Let `head_shard_block` be the result of running the fork choice on the assigned shard chain during the assigned slot.
+- Let `shard_blocks` be the shard blocks in the chain starting immediately _after_ the most recent crosslink (`head_state.shard_transitions[shard].latest_block_root`) up to the `head_shard_block`.
+
+*Note*: We assume that the fork choice only follows branches with valid `offset_slots` with respect to the most recent beacon state shard transition for the queried shard.
 
 ##### Head shard root
 
 Set `attestation_data.head_shard_root = hash_tree_root(head_shard_block)`.
 
 ##### Shard transition
 
-Set `shard_transition` to the value returned by `get_shard_transition()`.
+Set `shard_transition` to the value returned by `get_shard_transition(head_state, shard, shard_blocks)`.
 
 ```python
-def get_shard_transition(state: BeaconState,
+def get_shard_state_transition_result(
+    beacon_state: BeaconState,
+    shard: Shard,
+    shard_blocks: Sequence[SignedShardBlock],
+    validate_signature: bool=True,
+) -> Tuple[Sequence[ShardState], Sequence[Root], Sequence[uint64]]:
+    shard_states = []
+    shard_data_roots = []
+    shard_block_lengths = []
+
+    shard_state = beacon_state.shard_states[shard]
+    shard_block_slots = [shard_block.message.slot for shard_block in shard_blocks]
+    for slot in get_offset_slots(beacon_state, shard):
+        if slot in shard_block_slots:
+            shard_block = shard_blocks[shard_block_slots.index(slot)]
+            shard_data_roots.append(hash_tree_root(shard_block.message.body))
+        else:
+            shard_block = SignedShardBlock(message=ShardBlock(slot=slot))
```

Comment:
FYI I have a bug here that I will push it to shard fork choice rule PR to make tests pass:
```suggestion
            shard_block = SignedShardBlock(message=ShardBlock(slot=slot, shard=shard))
```
^^^^ need to set `shard` for empty proposal case.

Reply:
thanks!